packages/server/parseFormBody.ts

```diff
@@ -24,26 +34,34 @@ const parseRes = (res: HttpResponse) => {
 }
 
 const parseFormBody = async ({res, contentType}: ParseFormBodySignature): Promise<JSON | null> => {
+  let parsedBody: unknown
+  const parsedUploadables: UploadableBufferMap = {}
   const resBuffer = await parseRes(res)
   if (!resBuffer) return null
   const parts = getParts(resBuffer, contentType) as MultipartField[]
   if (!parts) return null
 
   try {
-    let parsedBody
     parts.forEach(({name, data, type}) => {
       if (name === 'body') {
         parsedBody = JSON.parse(Buffer.from(data).toString())
-        return
+      } else if (name.startsWith('uploadables')) {
+        const [, key] = name.split('.')
+        parsedUploadables[key] = {
+          contentType: type,
+          buffer: Buffer.from(data)
+        } as UploadableBuffer
       }
-      if (!name.startsWith('uploadables')) return
-      const [, key] = name.split('.')
-      parsedBody.payload.variables[key] = {
-        contentType: type,
-        buffer: Buffer.from(data)
-      } as Uploadable
     })
-    return parsedBody
+    if (!parsedBody || typeof parsedBody !== 'object') return null
+    if (Object.keys(parsedUploadables).length) {
+      const validParsedBody = parsedBody as FetchHTTPData
```

Comment:
what if the body is `'{}'`?
for all parsed JSON coming from an untrusted client we've gotta do runtime validation

Reply:
yea that's a good point. i was thinking to just throw the error and catch if the expected properties didn't exist in body. doing a runtime check is much more explicit though. i just added a user defined type guard, then type cast after the json is validated to be the correct shape.

it wasn't clear to me before the difference btwn `any` vs `unknown`, but now i see how it's useful using `unknown` in this case.